src/main/java/org/sagebionetworks/web/client/widget/discussion/SingleDiscussionThreadWidget.java

```diff
@@ -158,8 +159,28 @@ public void invoke() {
 		configureMessage();
 		configureReplies();
 		watchReplyCount();
+		invokeCheckForInViewAndLoadData = new Callback() {
+			@Override
+			public void invoke() {
+				checkForInViewAndLoadData();
+			}
+		};
+		checkForInViewAndLoadData();
 	}
-	
+
+	public void checkForInViewAndLoadData() {
+		if (!view.isLoadMoreAttached()) {
+			//Done, view has been detached and widget was never in the viewport
+			return;
+		} else if (view.isLoadMoreInViewport() && view.getLoadMoreVisibility()) {
+			//try to load data!
+			loadMore();
```

Comment:
If we get into loadMore once, what causes the code to recheck later (even if the loadMoreVisibility is set to true)?